server/autolinkapp/form.go

```diff
@@ -0,0 +1,277 @@
+	Scope                string            `json:"scope"`
+	WordMatch            bool              `json:"word_match"`
+	DisableNonWordPrefix bool              `json:"disable_non_word_prefix"`
+	DisableNonWordSuffix bool              `json:"disable_non_word_suffix"`
+
+	TestInput  string `json:"test_input"`
+	TestOutput string `json:"test_output"`
+
+	SubmitButtons string `json:"submit_buttons"`
+}
+
+func (a *app) handleFormFetch(w http.ResponseWriter, r *http.Request) {
+	call := &apps.CallRequest{}
+	err := json.NewDecoder(r.Body).Decode(call)
+	if err != nil {
+		httputils.WriteJSON(w, apps.NewErrorCallResponse(errors.Wrap(err, "failed to decode call request body")))
+		return
+	}
+
+	values, err := extractFormValues(call.Values)
+	if err != nil {
+		httputils.WriteJSON(w, apps.NewErrorCallResponse(errors.Wrap(err, "error extracting form values")))
+		return
+	}
+
+	if call.SelectedField == fieldLink {
+		if values.Link.Value == createOptionValue || values.Link.Value == "" {
+			httputils.WriteJSON(w, a.getEmptyFormResponse())
+			return
+		}
+
+		links := a.store.GetLinks()
+		for _, link := range links {
+			if link.Name == values.Link.Value {
+				values = formValuesFromAutolink(link)
+				break
+			}
+		}
+	}
+
+	f := a.getForm(values)
+	resp := apps.CallResponse{
+		Type: apps.CallResponseTypeForm,
+		Form: f,
+	}
+
+	httputils.WriteJSON(w, resp)
+}
+
+func (a *app) getForm(values FormValues) *apps.Form {
+	createNewOption := apps.SelectOption{
+		Label: "(create new)",
+		Value: createOptionValue,
+	}
+
+	links := a.store.GetLinks()
+	sort.Slice(links, func(i, j int) bool {
+		return strings.Compare(strings.ToLower(links[i].Name), strings.ToLower(links[j].Name)) < 0
+	})
+
+	linkOptions := []apps.SelectOption{createNewOption}
+	for _, link := range links {
+		linkOptions = append(linkOptions, apps.SelectOption{
+			Label: link.Name,
+			Value: link.Name,
+		})
+	}
+
+	linkOption := createNewOption
+	if values.Link.Value != "" {
+		linkOption = values.Link
+	}
+
+	return &apps.Form{
+		Title:         "Edit Autolinks",
+		SubmitButtons: fieldSubmitButtons,
+		Call:          apps.NewCall("/edit-links"),
+		Fields: []*apps.Field{
+			{
+				Name:                fieldLink,
+				ModalLabel:          "Link",
+				Type:                apps.FieldTypeStaticSelect,
+				Value:               linkOption,
+				SelectStaticOptions: linkOptions,
+				SelectRefresh:       true,
+			},
+			{
+				Name:       fieldName,
+				ModalLabel: "Name",
+				Type:       apps.FieldTypeText,
+				Value:      values.Name,
+				IsRequired: true,
+			},
+			{
+				Name:       fieldEnabled,
+				ModalLabel: "Enabled",
+				Type:       apps.FieldTypeBool,
+				Value:      values.Enabled,
+			},
+			{
+				Name:        fieldWordMatch,
+				ModalLabel:  "Word Match",
+				Type:        apps.FieldTypeBool,
+				Value:       values.WordMatch,
+				Description: "If true uses the [word boundaries](https://www.regular-expressions.info/wordboundaries.html)",
```

Comment:
Is there a ticket to render the MD?
![Screenshot from 2021-04-26 09-18-31](https://user-images.githubusercontent.com/16541325/116043834-72f2e100-a670-11eb-855e-c368bd3d8383.png)

Reply:
Yes there's a ticket to have a [markdown field](https://mattermost.atlassian.net/browse/MM-31052), but not one for a field's description. I tried to get this to work with our `Markdown` component, but it wasn't working for some reason.

Created ticket https://mattermost.atlassian.net/browse/MM-35179